Find system's Sqlite rather than version shipped with hashlink.

grepsuzette opened the pull request:
Lib is found by cmake and compiles on Linux, **need to test on Windows and OSX**

For future reference, there are various FindSqlite.cmake files in the wild, e.g.:
* https://github.com/LuaDist/libsqlite3/blob/master/cmake/FindSQLite3.cmake
* https://github.com/Kitware/CMake/blob/master/Modules/FindSQLite3.cmake
* https://github.com/WebKit/webkit/blob/master/Source/cmake/FindSqlite.cmake  (using this one)
* https://github.com/facebook/hhvm/blob/master/CMake/FindLibSQLite.cmake

This SO questions shows how tricky it can be to have one working for windows:
* https://stackoverflow.com/questions/15856445/cmake-find-sqlite3-library-on-windows

So let's hope this one works from the go; as said above it finds the system sqlite on Linux but apparently this is the easy part.

grepsuzette committed:
Find system's Sqlite rather than version shipped with hashlink.
Lib is found by cmake and compiles on Linux, need to test on Windows and OSX.

emugel commented:
If this works I suppose we will also want the same for mysql?

grepsuzette commented on libs/sqlite/CMakeLists.txt:
```diff
@@ -1,12 +1,14 @@
+
 add_library(sqlite.hdll SHARED
     sqlite.c
     ${INCLUDES_BASE_DIR}/sqlite/src/sqlite3.c
```
I wonder about that line though...

grepsuzette replied:
Probably worth asking @ncannasse here

grepsuzette replied:
Or perhaps @andyli is the cmake guru? :) 
Following Nicolas' recommendations we are trying to not use the sqlite shipped with hashlink but rather find it on the system, but I have a doubt whether keeping those lines unchanged is correct:

```
add_library(sqlite.hdll SHARED
    sqlite.c
    ${INCLUDES_BASE_DIR}/sqlite/src/sqlite3.c
```

jonasmalacofilho commented:
Thanks a lot @emugel , I'll take a look (probably tomorrow) and continue from there.

grepsuzette committed:
ensure libraries installed in CMAKE_INSTALL_PREFIX really get searched
there.

There is a mismatch noticed on Linux where the hdll
libraries are not found with a `hl` built with `cmake`. E.g. the
libraries get installed to `/usr/local/lib/*.hdll` while at runtime they
are searched in `/usr/lib/`.

This proposed change fixes the mismatch.
